examples/webserver/threadpool/src/main/java/io/helidon/examples/webserver/threadpool/GreetService.java

```diff
@@ -0,0 +1,212 @@
+
+    private static final Logger LOGGER = Logger.getLogger(GreetService.class.getName());
+
+    private static ExecutorService myThreadPool;
+
+    GreetService(Config config) {
+        greeting.set(config.get("app.greeting").asString().orElse("Ciao"));
+
+        // Build a thread pool using the configuration
+        myThreadPool = ThreadPoolSupplier.builder().config(config.get("application-thread-pool")).build().get();
+        if (myThreadPool instanceof ContextAwareExecutorService) {
+            ExecutorService wrapped = ((ContextAwareExecutorService) myThreadPool).unwrap();
+            LOGGER.info("Application thread pool: " + wrapped.toString());
+        }
+    }
+
+    /**
+     * A service registers itself by updating the routing rules.
+     * @param rules the routing rules.
+     */
+    @Override
+    public void update(Routing.Rules rules) {
+        rules
+            .get("/", this::getDefaultMessageHandler)
+            .get("/slowly/{name}", this::getMessageSlowlyHandler)
+            .get("/{name}", this::getMessageHandler)
+            .put("/greeting", this::updateGreetingHandler);
+
+    }
+
+    /**
+     * Return a worldly greeting message.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getDefaultMessageHandler(ServerRequest request, ServerResponse response) {
+        sendResponse(response, "World");
+    }
+
+    /**
+     * Return a greeting message using the name that was provided.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getMessageHandler(ServerRequest request, ServerResponse response) {
+        String name = request.path().param("name");
+        sendResponse(response, name);
+    }
+
+    /**
+     * Slowly Return a greeting message using the name that was provided.
+     * @param request the server request
+     * @param response the server response
+     */
+    private void getMessageSlowlyHandler(ServerRequest request, ServerResponse response) {
+
+        String name = request.path().param("name");
+
+        // One way to pass data to new thread is to use Context
+        Contexts.context().ifPresentOrElse(ctx -> ctx.register("NAME_PARAM", name + "_from_context"), () -> {
+                    throw new IllegalStateException("No context");
+                });
+
+        // Another way, just pass via Runnable.
+        myThreadPool.submit(() -> sendResponseSlowly(response, name, 3));
+    }
+
+    /**
+     * Send a response slowly. This simulates blocking business logic.
+     *
+     * @param response server response
+     * @param name name to greet
+     * @param sleepSeconds artificial delay to simulate blocking business logic
+     */
+    private void sendResponseSlowly(ServerResponse response, String name, int sleepSeconds) {
+
+        // Fetch NAME_PARAM from Context
+        String nameFromContext = Contexts.context()
```

Comment:
This is different, as we run in an executor, so this usage is correct

Reply:
Roger. Left as is.